rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/validation/CertificateTreeValidationService.java

```diff
@@ -319,19 +329,36 @@ private void validateCertificateAuthority(final TrustAnchor trustAnchor,
             }
 
             final X509Crl x509Crl = crl.get();
-            x509Crl.validate(crlUri.toASCIIString(), context, null, VALIDATION_OPTIONS, temporary);
+            x509Crl.validate(crlUri.toASCIIString(), context, null, validationOptions, temporary);
             if (temporary.hasFailureForCurrentLocation()) {
                 return;
             }
 
             temporary.setLocation(new ValidationLocation(manifestUri));
-            manifest.validate(manifestUri.toASCIIString(), context, x509Crl, manifest.getCrlUri(), VALIDATION_OPTIONS, temporary);
+            manifest.validate(manifestUri.toASCIIString(), context, x509Crl, manifest.getCrlUri(), validationOptions, temporary);
             if (temporary.hasFailureForCurrentLocation()) {
                 return;
             }
             validatedObjects.add(manifestObject.get().key());
 
-            manifest.getFiles().entrySet()
+
+            processManifest(trustAnchor, registeredRepositories, context, validatedObjects, temporary, manifestUri, manifest, crlUri, x509Crl);
+
+        }catch (StrictValidationException e) {
```

Comment:
Shouldn't it be 
try {
  processManifest(..)
} catch (StrictValidationException) {
}
instead of the whole method inside of the try {..}?
It would be much easier to undestand.

Reply:
Updated, I separate this processManifest originally to have process manifest strict vs process manifest non strict. But now this branch is happening on getManifestEntry and getValidationContext